pkgs/by-name/li/libcosmicAppHook/package.nix

```
@@ -0,0 +1,83 @@
+# SPDX-License-Identifier: MIT
+# SPDX-FileCopyrightText: Lily Foster <lily@lily.flowers>
+# Portions of this code are adapted from nixos-cosmic
+# https://github.com/lilyinstarlight/nixos-cosmic
+
+{
+  includeSettings ? true,
+  lib,
+  libGL,
+  libxkbcommon,
+  makeBinaryWrapper,
+  makeSetupHook,
+  pkg-config,
+  stdenv,
+  targetPackages,
+  vulkan-loader,
+  wayland,
+  xorg,
+}:
+
+makeSetupHook {
+  name = "libcosmic-app-hook";
+
+  propagatedBuildInputs = [
+    makeBinaryWrapper
+    pkg-config
+  ];
+
+  # ensure deps for linking below are available
+  depsTargetTargetPropagated =
+    assert (lib.assertMsg (!targetPackages ? raw) "libcosmicAppHook must be in nativeBuildInputs");
+    [
+      libGL
+      libxkbcommon
+      xorg.libX11
+      xorg.libXcursor
+      xorg.libXi
+      xorg.libxcb
+    ]
+    ++ lib.optionals (!stdenv.isDarwin) [
+      wayland
+      vulkan-loader
+    ];
+
+  substitutions = {
+    fallbackXdgDirs = "${lib.optionalString includeSettings "${targetPackages.cosmic-settings}/share:"}${targetPackages.cosmic-icons}/share";
+
+    cargoLinkerVar = stdenv.hostPlatform.rust.cargoEnvVarTarget;
+    # force linking for all libraries that may be dlopen'd by libcosmic/iced apps
+    cargoLinkLibs = lib.escapeShellArgs (
+      [
+        # for wgpu-hal
+        "EGL"
+        # for xkbcommon-dl
+        "xkbcommon"
+        # for x11-dl, tiny-xlib, wgpu-hal
+        "X11"
+        # for x11-dl, tiny-xlib
+        "X11-xcb"
+        # for x11-dl
+        "Xcursor"
+        "Xi"
+        # for x11rb
+        "xcb"
+      ]
+      ++ lib.optionals (!stdenv.isDarwin) [
+        # for wgpu-hal, wayland-sys
+        "wayland-client"
+        # for wgpu-hal
+        "wayland-egl"
+        "vulkan"
+      ]
+    );
+  };
+
+  meta = {
+    description = "Setup hook for configuring and wrapping applications based on libcosmic";
+    maintainers = with lib.maintainers; [
+      nyabinary
+      thefossguy
+    ];
```

I've been using COSMIC for a while and would be happy to help maintain it in nixpkgs. Feel free to add me to the maintainers list if you want to :)

```suggestion
    maintainers = with lib.maintainers; [
      nyabinary
      thefossguy
      HeitorAugustoLN
    ];
```

---

More maintainers is always better!